packages/beacon-node/src/chain/chain.ts

```diff
@@ -691,10 +699,48 @@ export class BeaconChain implements IBeaconChain {
 
   recomputeForkChoiceHead(): ProtoBlock {
     this.metrics?.forkChoice.requests.inc();
-    const timer = this.metrics?.forkChoice.findHead.startTimer();
+    const timer = this.metrics?.forkChoice.findHead.startTimer({entrypoint: FindHeadFnName.recomputeForkChoiceHead});
+
+    try {
+      return this.forkChoice.updateAndGetHead({mode: UpdateHeadOpt.GetCanonicialHead}).head;
+    } catch (e) {
+      this.metrics?.forkChoice.errors.inc();
+      throw e;
+    } finally {
+      timer?.();
+    }
+  }
+
+  predictProposerHead(slot: Slot): ProtoBlock {
+    this.metrics?.forkChoice.requests.inc();
+    const timer = this.metrics?.forkChoice.findHead.startTimer({entrypoint: FindHeadFnName.predictProposerHead});
+
+    try {
+      return this.forkChoice.updateAndGetHead({mode: UpdateHeadOpt.GetPredictedProposerHead, slot}).head;
+    } catch (e) {
+      this.metrics?.forkChoice.errors.inc();
+      throw e;
+    } finally {
+      timer?.();
+    }
+  }
+
+  getProposerHead(slot: Slot): ProtoBlock {
+    this.metrics?.forkChoice.requests.inc();
+    const timer = this.metrics?.forkChoice.findHead.startTimer({entrypoint: FindHeadFnName.getProposerHead});
+    const secFromSlot = this.clock.secFromSlot(slot);
 
     try {
-      return this.forkChoice.updateHead();
+      const {head, isHeadTimely, notReorgedReason} = this.forkChoice.updateAndGetHead({
+        mode: UpdateHeadOpt.GetProposerHead,
+        secFromSlot,
+        slot,
+      });
+      this.metrics?.forkChoice.isBlockTimely.set(isHeadTimely ? 1 : 0);
```

Comment:
A nice enhancement is to `RegistryMetricCreator` is to add support of boolean support to abstract the `bool ? 1 : 0` logic.

Reply:
I don't see a use case of adding boolean value to metrics like this, do you know if we have any other metrics that work like this?

I am having a hard time seeing how this metric produces meaningful data if you just set a gauge to 0/1, have you tried to visualize this in grafana?

What data points are you interested in? guess somehow correlating timely blocks with not reorged reason?

Reply:
> What data points are you interested in? guess somehow correlating timely blocks with not reorged reason?

Yes, eg. If the head block is not timely, what was the reason why it wasn't re-orged

Reply:
> > What data points are you interested in? guess somehow correlating timely blocks with not reorged reason?
> 
> Yes, eg. If the head block is not timely, what was the reason why it wasn't re-orged

The data collected by metrics is cumulative, if you want to look at a single data point it is better to add a log for this.

We have metrics that are used to display the current state of the system which uses boolean-like values to do that.
https://github.com/ChainSafe/lodestar/blob/7fadd303fad8679ee2b7462747b9d9fc33dafb23/packages/validator/src/metrics.ts#L280-L282


For what you are trying to achieve it's better to use an incrementing counter (gauge) and add a label with reason.

Reply:
> For what you are trying to achieve it's better to use an incrementing counter (gauge) and add a label with reason.

Ah this is helpful. Will change accordingly thanks